From 1073f0e792d9d5bf9d53bddb6e51177c8024c2b3 Mon Sep 17 00:00:00 2001
From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Thu, 29 Jun 2017 10:12:46 -0400
Subject: data: Add gnome-login.session

This reverts
https://git.gnome.org/browse/gdm/commit/?id=f66cdfcb2
and renames the session from gdm-shell to gnome-login.

The removal caused problems for Ubuntu 17.10 which does not
include gnome.session by default.

https://launchpad.net/bugs/1701243

https://bugzilla.gnome.org/show_bug.cgi?id=784340
---
 data/Makefile.am                         | 12 ++++++++++++
 data/dconf/defaults/00-upstream-settings |  2 +-
 data/gnome-login.session.in              |  3 +++
 3 files changed, 16 insertions(+), 1 deletion(-)
 create mode 100644 data/gnome-login.session.in

diff --git a/data/Makefile.am b/data/Makefile.am
index f2875fc..c10c915 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -86,9 +86,20 @@ gdm.schemas.in: $(srcdir)/gdm.schemas.in.in
 		-e 's,[@]sbindir[@],$(sbindir),g' \
 		<$(srcdir)/gdm.schemas.in.in >gdm.schemas.in
 
+
+EXTRA_DIST += gnome-login.session.in
+
+gnome-login.session: $(srcdir)/gnome-login.session.in
+	sed	-e 's,[@]libexecdir[@],$(libexecdir),g' \
+		-e 's,[@]CHECK_ACCELERATED_DIR[@],$(CHECK_ACCELERATED_DIR),g' \
+		< $< > $@.tmp && mv $@.tmp $@
+
 localealiasdir = $(datadir)/gdm
 localealias_DATA = locale.alias
 
+sessiondir = $(datadir)/gnome-session/sessions
+session_DATA =  gnome-login.session
+
 pam_redhat_files = \
 	pam-redhat/gdm-autologin.pam		\
 	pam-redhat/gdm-launch-environment.pam	\
@@ -179,6 +190,7 @@ CLEANFILES = 				\
 DISTCLEANFILES =			\
 	$(dbusconf_DATA)		\
 	gdm.schemas			\
+	gnome-login.session		\
 	$(NULL)
 
 MAINTAINERCLEANFILES =			\
diff --git a/data/dconf/defaults/00-upstream-settings b/data/dconf/defaults/00-upstream-settings
index c0b3cc6..175d661 100644
--- a/data/dconf/defaults/00-upstream-settings
+++ b/data/dconf/defaults/00-upstream-settings
@@ -7,7 +7,7 @@
 #
 
 [org/gnome/desktop/session]
-session-name='gnome'
+session-name='gnome-login'
 
 [org/gnome/desktop/background]
 show-desktop-icons=false
diff --git a/data/gnome-login.session.in b/data/gnome-login.session.in
new file mode 100644
index 0000000..278d8d4
--- /dev/null
+++ b/data/gnome-login.session.in
@@ -0,0 +1,3 @@
+[GNOME Session]
+Name=Display Manager
+RequiredComponents=org.gnome.Shell;org.gnome.SettingsDaemon.A11yKeyboard;org.gnome.SettingsDaemon.A11ySettings;org.gnome.SettingsDaemon.Clipboard;org.gnome.SettingsDaemon.Color;org.gnome.SettingsDaemon.Datetime;org.gnome.SettingsDaemon.Housekeeping;org.gnome.SettingsDaemon.Keyboard;org.gnome.SettingsDaemon.MediaKeys;org.gnome.SettingsDaemon.Mouse;org.gnome.SettingsDaemon.Orientation;org.gnome.SettingsDaemon.Power;org.gnome.SettingsDaemon.PrintNotifications;org.gnome.SettingsDaemon.Rfkill;org.gnome.SettingsDaemon.ScreensaverProxy;org.gnome.SettingsDaemon.Sharing;org.gnome.SettingsDaemon.Smartcard;org.gnome.SettingsDaemon.Sound;org.gnome.SettingsDaemon.Wacom;org.gnome.SettingsDaemon.XRANDR;org.gnome.SettingsDaemon.XSettings;
-- 
cgit v0.12
